spark/sql/session/sparksession.go

```diff
@@ -49,21 +51,27 @@ func (s *SparkSessionBuilder) Remote(connectionString string) *SparkSessionBuild
 	return s
 }
 
-func (s *SparkSessionBuilder) Build(ctx context.Context) (SparkSession, error) {
+func (s *SparkSessionBuilder) ChannelBuilder(cb channel.Builder) *SparkSessionBuilder {
+	s.channelBuilder = cb
+	return s
+}
 
-	cb, err := channel.NewBuilder(s.connectionString)
-	if err != nil {
-		return nil, sparkerrors.WithType(fmt.Errorf("failed to connect to remote %s: %w", s.connectionString, err), sparkerrors.ConnectionError)
+func (s *SparkSessionBuilder) Build(ctx context.Context) (SparkSession, error) {
```

Comment:
I would have made builder a argument to build:
```golang
func (s *SparkSessionBuilder) Build(ctx context.Context, cb channel.Builder) (SparkSession, error) {
```

Reply:
I guess if you want a builder pattern, then the natural way would be to have a `WithChannelBuilder` call

Reply:
I renamed the `ChannelBuilder` method to `WithChannelBuilder`